drivers/serial/uart_npcx.c

```diff
@@ -52,6 +56,26 @@ struct uart_npcx_data {
 #define HAL_INSTANCE(dev) \
 	(struct uart_reg *)(DRV_CONFIG(dev)->uconf.base)
 
+#if defined(CONFIG_PM) && defined(CONFIG_UART_INTERRUPT_DRIVEN)
+static void uart_npcx_pm_constraint_set(const struct device *dev)
+{
+	struct uart_npcx_data *data = DRV_DATA(dev);
+
+	if (atomic_set(&data->pm_constraint_on, 1) == 0) {
+		pm_constraint_set(PM_STATE_SUSPEND_TO_IDLE);
+	}
```

Comment:
note: the PM constraint API uses ref counting, are you sure you need atomics here? If you call `pm_constraint_set` N times, the constraint won't be released after N other calls to `pm_constraint_release`.

Reply:
We use the constrain for UART transmission. pm_constraint_set() may be invoked when any UART transmission request. However, we couldn't know how much time invoking pm_constraint_set() if we didn't maintain a local counter when the complete transmission ISR. The pm_constraint_set() may be invoked several times before the complete ISR. So we use this to guarantee the set/release is compared.